### ISSUE TITLE
LDAP Login with multiple group mappings not working

### ISSUE THREAD

**Issue by virtualdreams** [grafana/grafana#87046]
### What happened?

Login fails if multiple `[[servers.group_mappings]]` are used.

### What did you expect to happen?

Loggin and group mappings are working.

### Did this work before?

Without defining an admin group, the login works without any problems.

### How do we reproduce it?

Domain, Host and Password redacted.

This works:

```
[[servers]]
host = "dc-rz.example.com"
port = 389

# Search user bind dn
bind_dn = "CN=ldap-svcauth,OU=Funktion,OU=AD-Management,DC=example,DC=com"
bind_password = "Password"

# User search filter, for example "(cn=%s)" or "(sAMAccountName=%s)" or "(uid=%s)"
search_filter = "(sAMAccountName=%s)"

# An array of base dns to search through
search_base_dns = ["DC=example,DC=com"]

# Specify names of the ldap attributes your ldap uses
[servers.attributes]
name = "givenName"
surname = "sn"
username = "sAMAccountName"
member_of = "memberOf"
email =  "email"

[[servers.group_mappings]]
# If you want to match all (or no ldap groups) then you can use wildcard
group_dn = "*"
org_role = "Viewer"
org_id = 1
```

Not working:

```
[[servers]]
host = "dc-rz.example.com"
port = 389

# Search user bind dn
bind_dn = "CN=ldap-svcauth,OU=Funktion,OU=AD-Management,DC=example,DC=com"
bind_password = "Password"

# User search filter, for example "(cn=%s)" or "(sAMAccountName=%s)" or "(uid=%s)"
search_filter = "(sAMAccountName=%s)"

# An array of base dns to search through
search_base_dns = ["DC=example,DC=com"]

# Specify names of the ldap attributes your ldap uses
[servers.attributes]
name = "givenName"
surname = "sn"
username = "sAMAccountName"
member_of = "memberOf"
email =  "email"

# new mapping for admins
[[servers.group_mappings]]
group_dn = "CN=grafana-admins,OU=Monitoring,OU=Tools und Technik,OU=AD-Management,DC=example,DC=com"
org_role = "Admin"
org_role = 1

[[servers.group_mappings]]
# If you want to match all (or no ldap groups) then you can use wildcard
group_dn = "*"
org_role = "Viewer"
org_id = 1
```

Error:

```logger=authn.service t=2024-04-29T14:14:14.01305408+02:00 level=info msg="Failed to authenticate request" client=auth.client.form error="[password-auth.failed] failed to authenticate identity: unable to create LDAP client\n[password-auth.invalid] invalid password"
logger=context userId=0 orgId=1 uname= t=2024-04-29T14:14:14.013149193+02:00 level=info msg=Unauthorized error="[password-auth.failed] failed to authenticate identity: unable to create LDAP client\n[password-auth.invalid] invalid password" remote_addr=PRIVATEIP traceID=

```

### Environment (with versions)?

Grafana: 10.4.2
OS: Debian 12
Browser: Firefox

### Grafana platform?

A package manager (APT, YUM, BREW, etc.)

**Comment by virtualdreams** [comment]
Find a duplicate line in one mapping that caused the error:

```
org_role = "Admin"
org_role = 1
```

**Comment by virtualdreams** [comment]
> Works fine for us
> 
> [[servers.group_mappings]] group_dn = "CN=Grafana-Admins-Group.....,DC=com" org_role = "Admin" grafana_admin = true # Available in Grafana v5.3 and above
> 
> [[servers.group_mappings]] group_dn = "CN=Grafana-Viewers-Only-....DC=com" org_role = "Viewer" org_id = 1
> 
> [[servers.group_mappings]] group_dn = "CN=Grafana-Viewers-ORg2....DC=com" org_role = "Viewer" org_id = 2

I had a typo in my configuration and didn't see it.

It still doesn't work because it can't resolve the group members correctly as there is another group in the group.

My login name is: `adm.username`

```
cn=adm.username-tmp,ou=admins-lit,ou=ad-management,dc=example,dc=com No match 
```
